frontend/public/components/utils/file-input.tsx

```diff
@@ -151,25 +167,33 @@ export const DroppableFileInput = withDragDropContext(
         });
         return;
       }
+      let inputFileIsBinary = false;
       const reader = new FileReader();
       reader.onload = () => {
         const input = reader.result as string; // Note(Yaacov): we use reader.readAsText
-        this.setState(
-          {
-            inputFileName: file.name,
-            inputFileData: input,
-            errorMessage: '',
-          },
-          () => this.props.onChange(input),
-        );
+        if (containsNonPrintableCharacters(input) && !inputFileIsBinary) {
+          inputFileIsBinary = true;
+          reader.readAsBinaryString(file);
+        } else {
+          this.setState(
+            {
+              inputFileName: file.name,
+              inputFileData: inputFileIsBinary ? Base64.encode(input) : input,
+              inputFileIsBinary,
```

Comment:
We need to handle this in the caller if we're not already. If this is the import YAML page, we should show an error saying the file contains unprintable characters. For secrets, we should allow the value but show a message saying the file contains unprintable characters and not preview the content.

Reply:
@spadgett Yes, the import yaml page indeed shows the message you mention above. I added the message about non printable characters right in the file-input.